test/catch2/live_restore/api/test_live_restore_fs_remove_rename.cpp

```diff
@@ -0,0 +1,117 @@
+    std::string dest_filename2 = env.dest_file_path("file2");
+    create_file(source_filename2);
+    REQUIRE(fs->fs_remove(fs, wt_session, dest_filename2.c_str(), 0) == 0);
+    // Ensure we didn't remove the source file.
+    REQUIRE(testutil_exists(nullptr, source_filename2.c_str()));
+    REQUIRE(check_stop(dest_filename2));
+    // Ensure we didn't create a stop file in the source.
+    REQUIRE(!check_stop(source_filename2));
+
+    // We can recreate a file with the same name as the one we removed earlier and remove it again.
+    create_file(dest_filename2);
+    REQUIRE(fs->fs_remove(fs, wt_session, dest_filename2.c_str(), 0) == 0);
+    REQUIRE(check_stop(dest_filename2));
+}
+
+TEST_CASE("Live Restore fs_rename", "[live_restore],[live_restore_remove_rename]")
+{
+    /*
+     * Note: this code runs once per SECTION so we're creating a brand new WT database for each
+     * section. If this gets slow we can make it static and manually clear the destination and
+     * source for each section.
+     */
+    live_restore_test_env env;
+
+    WTI_LIVE_RESTORE_FS *lr_fs = env.lr_fs;
+    WT_FILE_SYSTEM *fs = &lr_fs->iface;
+    WT_SESSION *wt_session = reinterpret_cast<WT_SESSION *>(env.session);
+
+    // Rename a file that only exists in the destination.
+    std::string dest_filename = env.dest_file_path("file");
+    std::string dest_rename = env.dest_file_path("file_rename");
+    create_file(dest_filename);
+    REQUIRE(fs->fs_rename(fs, wt_session, dest_filename.c_str(), dest_rename.c_str(), 0) == 0);
+    REQUIRE(check_stop(dest_filename));
+    REQUIRE(check_stop(dest_rename));
+    REQUIRE(!testutil_exists(nullptr, dest_filename.c_str()));
+    REQUIRE(testutil_exists(nullptr, dest_rename.c_str()));
+
+    // Renaming a file that doesn't exist fails.
+    REQUIRE(lr_fs->os_file_system->fs_rename(lr_fs->os_file_system, wt_session,
+              dest_filename.c_str(), dest_rename.c_str(), 0) == ENOENT);
+    REQUIRE(fs->fs_rename(fs, wt_session, dest_filename.c_str(), dest_rename.c_str(), 0) == ENOENT);
+
+    // Renaming a file that only exists in the source fails as we require that the file exists in
+    // the destination.
+    std::string source_filename = env.source_file_path("file2");
+    std::string dest_filename2 = env.dest_file_path("file2");
+    create_file(source_filename);
+    // Note: We need to pass dest filename here as WiredTiger would only refer to files as existing
+    // in the home directory.
+    REQUIRE(
+      fs->fs_rename(fs, wt_session, dest_filename2.c_str(), dest_rename.c_str(), 0) == EINVAL);
+
+    // Renaming over the top of an existing file succeeds.
+    REQUIRE(testutil_exists(nullptr, dest_rename.c_str()));
+    create_file(dest_filename2);
+    REQUIRE(fs->fs_rename(fs, wt_session, dest_filename2.c_str(), dest_rename.c_str(), 0) == 0);
+    REQUIRE(!testutil_exists(nullptr, dest_filename.c_str()));
+    REQUIRE(testutil_exists(nullptr, dest_rename.c_str()));
+}
```

Comment:
Same comment here about tests when we're in the COMPLETE state.

Reply:
bf0b231452c130ef62d31d5058e889ae3be13905